pkgs/applications/misc/hugo/010-call-rst2html-directly.patch

```diff
@@ -0,0 +1,51 @@
+From: Shreyansh Khajanchi <shreyansh_k@live.com>
+Date: Wed, 5 Sep 2018 20:06:25 +0530
+Subject: helpers/content.go: call rst2html directly to render content
+
+Hugo launches rst2html by calling Python interpreter and passing it the path of rst2html file.
+This doesn't work on NixOS because the real rst2html is actually wrapped behind a bash script which is responsible for correctly setting the env and then exec-ing the real rst2html.
+
+The problem is that rst2html is actually a bash script and not a Python script. Hence, Python fails to launch it with SyntaxError.
+
+This patch ensures that rst2html is called directly so the shebang work correctly.
+---
+ helpers/content.go | 16 ++--------------
+ 1 file changed, 2 insertions(+), 14 deletions(-)
+
+diff --git a/helpers/content.go b/helpers/content.go
+index 55d8ce202..18c648a43 100644
+--- a/helpers/content.go
++++ b/helpers/content.go
+@@ -664,21 +664,9 @@ func getRstExecPath() string {
+ 	return path
+ }
+ 
+-func getPythonExecPath() string {
+-	path, err := exec.LookPath("python")
```

Comment:
This could be even incorrect if `python` is not the correct version python version for `rst2html`. Can you also open an issue upstream? They really only should do this on windows in my opinion and even there this probably also will work if rst2html is associated with the correct extension (i.e. rst2html.py).

Reply:
Also they check if `rst2html` is in `$PATH` and is executable. Is this even true on windows?

Reply:
After digging in LookPath on windows it appears that .py has to be in `PATHEXT` to be found by LookPath. This means if it is not already a registered extension it would not be found anyway. If it is registered then prepending python should be not necessary.

Reply:
I know. I am unable to follow their reasoning as well.
I opened an issue upstream like a month ago but haven't received any response at all.
I think it's maybe because Nix is not on their priority list as they have quite a many other issues open. So, maybe we can apply the patch ourselves.

Of all the platforms where Nix works, except Windows, all natively support shebangs.
And, on Windows, Nix works under WSL (experimental) which supports shebangs. So, I believe, we can directly call rst2html for our use case without any issues.

Reply:
Additionally for reference: https://github.com/gohugoio/hugo/issues/5167
Hopefully the devs will find a solution.

Reply:
Ok. Please leave the pull request url as commend above the patch and we go merging it until a better solution is found.

Reply:
Just did that.
I think I also found a solution (https://github.com/gohugoio/hugo/pull/5285).

Reply:
Ideally, it should be merged. Hugo's next release is at least a month away.

Reply:
The idea was to use `fetchpatch` instead.

Reply:
Oh! let's wait for upstream to merge it.